server/stdio.go

```diff
@@ -189,39 +189,23 @@ func (s *StdioServer) processInputStream(ctx context.Context, reader *bufio.Read
 // returns an empty string and the context's error. EOF is returned when the input
 // stream is closed.
 func (s *StdioServer) readNextLine(ctx context.Context, reader *bufio.Reader) (string, error) {
-	readChan := make(chan string, 1)
-	errChan := make(chan error, 1)
-	done := make(chan struct{})
-	defer close(done)
+	type result struct {
+		line string
+		err  error
+	}
+
+	resultCh := make(chan result, 1)
 
 	go func() {
-		select {
-		case <-done:
-			return
-		default:
-			line, err := reader.ReadString('\n')
-			if err != nil {
-				select {
-				case errChan <- err:
-				case <-done:
-				}
-				return
-			}
-			select {
-			case readChan <- line:
-			case <-done:
-			}
-			return
-		}
+		line, err := reader.ReadString('\n')
+		resultCh <- result{line: line, err: err}
 	}()
 
 	select {
 	case <-ctx.Done():
 		return "", ctx.Err()
```

Comment:
if the context is be cancelled, the line 169 will get a context cancel error and print `Error reading input: context canceled` message, so it maybe ignore context cancel at line 169?

Reply:
Just asking, but shouldn't be the caller's responsibility to check the nature of the error that decide what to log and what no to? I'm removing it either way, this is taking to long to merge and many people cannot upgrade to v0.30.0 just because of this issue.